build_cmake/Framework/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+cmake_minimum_required(VERSION 2.8)
+project(Framework)
+
+set(CMAKE_MODULE_PATH
+	${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/cmake-modules
+	${CMAKE_MODULE_PATH}
+)
+
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
No, my initial understanding was that `CMAKE_BUILD_TYPE` didn't influence the preprocessor flags, but reading their [wiki](https://cmake.org/Wiki/CMake_Useful_Variables#Compilers_and_Tools) now seems to suggest otherwise, so long as you exclude `MSVC` since it sets these values already and it uses slight different argument style `/D` vs `-D`